prov/verbs/src/fi_verbs.c

```diff
@@ -2405,10 +2489,26 @@ fi_ibv_open_ep(struct fid_domain *domain, struct fi_info *info,
 		connreq = container_of(info->handle, struct fi_ibv_connreq, handle);
 		_ep->id = connreq->id;
 		free(connreq);
+        } else if (info->handle->fclass == FI_CLASS_PEP) {
+		pep = container_of(info->handle, struct fi_ibv_pep, pep_fid.fid);
+		_ep->id = pep->id;
```

Comment:
You should NULL out the cm_id of the passive endpoint, so that `fi_close()` can then be called on the passive endpoint without breaking the new active endpoint created from the passive endpoint.  Otherwise, you have a resource leak since the passive endpoint can't ever be freed.

@shefty Maybe this should be cleared up in the man page as well.  Adding something like this to the end of the last paragraph in the `fi_endpoint / fi_scalable_ep / fi_passive_ep` section?

> Once an active endpoint acquires the properties of a passive endpoint, the passive endpoint is no longer bound to any fabric resources and must no longer be used.  The user is expected to close the passive endpoint after opening the active endpoint in order to free up any lingering resources that had been used.

The alternative is to automatically close the passive endpoint when the active endpoint is created, but that may be unfriendly to language bindings, which would have to track whether or not they have to close the passive endpoint (I'm thinking of C++ in particular).